Add gotmpl tool (#280)
open-telemetry/opentelemetry-go-build-tools: gotmpl/README.md

# gotmpl

> `gotmpl` is a tool that uses Go's [`text/template`](https://pkg.go.dev/text/template)

Reviewer: ```suggestion
`gotmpl` is a tool
```

> for generting generating text files

Reviewer: ```suggestion
to generate files
```

> based on file templates and input JSON data.

Reviewer: ```suggestion
from [templates](https://pkg.go.dev/text/template) and JSON data.
```

## Usage

```sh
Usage of gotmpl:
  -b, --body string   Template body filepath.
  -d, --data string   Data in JSON format.
  -o, --out string    Output filepath.
```

## Background

`gotmpl` has been created as one of the solutions
to avoid depending on internal packages of different modules.
Such dependencies could transitively lead to build failure
when an internal package API introduces a non-backward compatible change.

> `gotmpl` can be used for sharing non-exported,

Reviewer: ```suggestion
`gotmpl` can be used for sharing non-exported
```
